plugin/mutate/source/dextool/plugin/mutate/backend/report/html/page_tree_map.d

```diff
@@ -0,0 +1,68 @@
+module dextool.plugin.mutate.backend.report.html.page_tree_map;
+
+import arsd.dom : Document, Element, require, Table, RawSource;
+
+import dextool.plugin.mutate.backend.report.html : FileIndex;
+import dextool.plugin.mutate.backend.report.html.tmpl : tmplBasicPage;
+
+@trusted auto makeTreeMapJSON(FileIndex[] files) {
+    import std.stdio : writeln;
+    import std.array : array, split;
+    import std.range : enumerate;
+    import std.json : JSONValue;
+
+    JSONValue root = ["name" : "root"];
+    auto curr = root;
+    foreach (f; files) {
+        auto path = f.display.split("/");
+
+        foreach (i, seg; path.enumerate) {
```

Comment:
This is a bit hard to understand. I think my problem is that the code is both dependent on adding data to JSONValue and looking what has been added mixed in with an uneven control flow.
It would be easier, I think, if the first part transformed the data to an intermediary structure as "simple and straight" as possible. Trying to keep the control flow continuous.
Then the intermediary is converted to json.